main.tf

```diff
@@ -10,3 +10,42 @@ resource "ibm_resource_instance" "scc_instance" {
   resource_group_id = var.resource_group_id
   tags              = var.resource_tags
 }
+
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+resource "ibm_iam_authorization_policy" "scc_cos_s2s_access" {
+  count                       = var.skip_cos_iam_authorization_policy ? 0 : 1
+  source_service_name         = "compliance"
+  source_resource_instance_id = ibm_resource_instance.scc_instance.guid
+  roles                       = ["Writer"]
+
+  resource_attributes {
+    name     = "serviceName"
+    operator = "stringEquals"
+    value    = "cloud-object-storage"
+  }
+
+  resource_attributes {
+    name     = "serviceInstance"
+    operator = "stringEquals"
+    value    = var.cos_instance_crn
+  }
+
+  resource_attributes {
+    name     = "accountId"
+    operator = "stringEquals"
+    value    = data.ibm_iam_account_settings.iam_account_settings.account_id
+  }
+}
+
```

Comment:
Did you test if event notifications are working? Don't we need a s2s auth policy for it to work?

Reply:
yes, I've tested integrating event notifications service and it works. 
No S2S auth policy required when setting up using terraform but auth policy is required when integrating from UI. Not sure if this is intended.

Reply:
I can't see how this will work without an auth policy. Are you sure there was not one maybe already in our account that allowed this to work?
Can you show proof that event notifications are working?
From what I am reading, an IAM authorization between Security and Compliance Center and Event Notifications (Event Source Manager role) is required for this to work, so our module needs to support optionally creating it.